File: MyViewModelDemo/app/src/main/java/com/example/myviewmodeldemo/MainActivity.java
package com.example.myviewmodeldemo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;

import com.example.myviewmodeldemo.databinding.MainActivityBinding;

public class MainActivity extends AppCompatActivity {

    private MainActivityBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = MainActivityBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);
    }

}

File: tryRadio/app/src/main/java/com/example/tryradio/MainActivity.kt
package com.example.tryradio

import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.RadioGroup
import com.example.tryradio.ui.main.MainFragment
import com.example.tryradio.databinding.MainActivityBinding
import com.example.tryradio.ui.main.MoreInfoFrag

class MainActivity : AppCompatActivity(),
               MoreInfoFrag.OnFragmentInteractionListener {

    private lateinit var binding: MainActivityBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = MainActivityBinding.inflate(layoutInflater)
        setContentView(binding.root)

    }

    override fun onFragmentInteraction(uri: Uri) {

    }
}